.github/workflows/Tests.yaml

```diff
@@ -0,0 +1,221 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+# Continuous integration tests that pull requests are required to pass
+name: Tests
+
+on: [push, pull_request]
+
+jobs:
+  # Make sure no commits are prefixed with `fixup` or similar keywords. See
+  # `tools/CheckCommits.sh` for details.
+  check_commits:
+    name: Commits
+    # Only run on pull requests since we don't check _all_ commits, but only
+    # those that came after the PR's base ref.
+    if: github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v1
+      - name: Check commits
+        # `CheckCommits.sh` tests against the local `develop` branch, so that's
+        # where we fetch the pull-request's base-branch to. Typically, it is
+        # the upstream `sxs-collaboration/spectre/develop` branch.
+        run: >
+          cd $GITHUB_WORKSPACE
+
+          git remote add upstream
+          https://github.com/${{ github.repository }}.git
+
+          git remote -v
+
+          git fetch upstream ${{ github.base_ref }}:develop
+
+          ./tools/CheckCommits.sh
+
+  # Run simple textual checks over files in the repository, e.g. checking for
+  # a license, line length limits etc. See `tools/CheckFiles.sh` for details.
+  check_files:
+    name: Files
+    # No need to do this again when opening a PR, so only check on `push`
```

Comment:
I added a comment to the very beginning of the file, where the `on` configuration is set. Please let me know if that's clear

Reply:
Okay, that helps a lot! One thing that's still not clear: does opening a PR send both a `push` and a `pull_request` event? What happens when a PR is updated?

Reply:
I think the thing that is not clear to me is what is a pull_request event and what is a push event? How does this interact with (i) opening a PR (ii) pushing to a PR (iii) merging into develop?

Reply:
Pushing to the branch triggers the `push` event, and both opening _and_ pushing to an already-opened PR triggers the `pull_request` event. See [these docs](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#pull-request-event-pull_request) for details on the `pull_request` event.

I reconsidered the triggers and edited the `on` configuration at the beginning of the file. The idea is to run everything on PRs, and to also test merge commits, but not run everything on each push to a branch that doesn't have a PR open. One subtlety is that by running unit tests on `pull_request` events instead of `push` events, they get access to the caches on the PR's base branch now.

Reply:
Okay, I still don't really understand. I'll read through the documentation you linked, but it highlights my concern: it is not at all clear when what checks are run. With the current travis config nothing is run on user forks by default, they have to set that up. All checks are run all the time when commits are pushed to a PR and when anything is merged into develop. I think you should make a clear bullet list of what is happening and add that as a comment to the file.

Reply:
Perhaps you missed this note in the documentation:

> Note: By default, a workflow only runs when a `pull_request`'s activity type is `opened`, `synchronize`, or `reopened`. To trigger workflows for more activity types, use the `types` keyword.

The Travis setup you describe is precisely how this GitHub workflow should behave (if it works the way it is described in the documentation):
- Nothing will run on forks unless users enable it.
- All checks run on opening, updating and re-opening PRs, with access to cache from the base repository.
- All checks except those that require a PR (clang-tidy and CheckCommits need a base ref to compare commits to) run when merge-commits are pushed to `develop`.